### DISCUSSION TITLE
Preconnect to required origins

### DISCUSSION THREAD

**Discussion by anotheri** [yeskunall/astro-umami#15] [Ideas]
It would be nice to have the `hostUrl` automatically pre-connected, if _umami_ is going to be used (i.e. if valid `id` provided).

As per Lighthouse report:

![image](https://github.com/user-attachments/assets/195327a5-41ea-44dc-b3d6-899861eaa9b9)

Read more: https://developer.chrome.com/docs/lighthouse/performance/uses-rel-preconnect/?utm_source=lighthouse&utm_medium=devtools

**Comment by yeskunall** [comment]
Ah, good point. We use our own hosted version of Umami so the need never arose.

I’m willing to add this, but with a section to the README outlining the privacy implications (if not concerns) behind enabling either of those two.

My suggestion would be to instead self-host Umami, and then use [`rel=preload`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preload) to preload the tracking script.

**Comment by anotheri** [comment]
Yeah `preload` should work even better i assume. In my tests adding `<link rel="preconnect" href={hostUrl} />` helped (so browser knows that something will happen with this host).
